Cargo.toml

```diff
@@ -2980,6 +2981,14 @@ description = "Demonstrates customizing default window settings"
 category = "Window"
 wasm = true
 
+[[example]]
```

Comment:
What happens when this is removed? We shouldn't be listing tests as examples TBH.

Reply:
That surprised me too, but I kept the behaviour as closely as what's existing (so in my opinion this is out of scope for this Pull Request):

Compare this to next example in Cargo.toml: `resizing`, which is used as a test, to be noted it has a metadata `hidden = true`.

If we remove that block, we cannot run the example 😅, mostly a blocker for [ci runs](https://github.com/Vrixyz/bevy/blob/63937a77d2144b1e95f29c6046e393e8f1689a93/.github/workflows/ci.yml#L246).

I believe using examples for testing is a workaround to be guaranteed main thread execution, because if I'm trying to run my example as a test: I have the following error:

```sh
$ cargo test --test ambiguity_detection
   Compiling bevy v0.15.0-dev (C:\Users\thier\Documents\perso\bevy)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 17.69s
     Running tests/ecs/ambiguity_detection.rs (C:\Users\thier\Documents\cargo_target_dir\debug\deps\ambiguity_detection-eeb84a8dea010d13.exe)

running 1 test
2024-07-17T14:43:32.965951Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Windows 10 Home", kernel: "19045", cpu: "AMD Ryzen 5 2600X Six-Core Processor", core_count: "6", memory: "32.0 GiB" }
test ambiguity_detection ... FAILED

failures:

---- ambiguity_detection stdout ----
thread 'ambiguity_detection' panicked at C:\Users\thier\.cargo\registry\src\index.crates.io-6f17d22bba15001f\winit-0.30.2\src\platform_impl\windows\event_loop.rs:180:13:   
Initializing the event loop outside of the main thread is a significant cross-platform compatibility hazard. If you absolutely need to create an EventLoop on a different thread, you can use the `EventLoopBuilderExtWindows::any_thread` function.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
I've tried one last thing, it seems possible to run those as `test` if we bypass the harness:

```patch
diff --git a/Cargo.toml b/Cargo.toml
index d6eefaa91..533198ce6 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2981,12 +2981,13 @@ description = "Demonstrates customizing default window settings"
 category = "Window"
 wasm = true
 
-[[example]]
+[[test]]
 name = "ambiguity_detection"
 path = "tests/ecs/ambiguity_detection.rs"
 doc-scrape-examples = true
+harness = false
 
-[package.metadata.example.ambiguity_detection]
+[package.metadata.test.ambiguity_detection]
 hidden = true
 
 [[example]]
```

I'm not sure about its implications on the rest of bevy machinery though (example scraping, website building, ci tests...), so I'd be happy to study that in a follow up PR rather than in this one :)